osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs

```diff
@@ -60,7 +62,14 @@ protected override void Update()
         {
             base.Update();
 
-            if (!IsBreakTime.Value)
+            if (IsBreakTime.Value)
+            {
+                breakend = false;
+                return;
+            }
+
+            // Health should only be drained after the first post-Break Hit is judged
+            if (!IsBreakTime.Value && breakend)
```

Comment:
The `!IsBreakTime.Value` check is now redundant, as the method early-returns if it's true (so it can't *not* be false at this point).

After eliminating it you could as well negate the other check, and I *think* the following variant is equivalent:
```csharp
// can check this first, as drain is re-enabled externally by ApplyResultInternal
if (!applyDrain)
    return;

// drain is on - check if a break is starting; if yes, disable and early-return
if (IsBreakTime.Value)
{
    applyDrain = false;
    return;
}

// actually apply drain
```

Reply:
Fixed in e09786e.